### PR TITLE
Adding workaround for `rad env init kubernetes` on EKS

### PR DESCRIPTION
**Pull request by willdavsmith**
* Added info for workaround for `rad env init kubernetes` on EKS
* Removing `--fargate` from EKS cluster initialization steps

Closes: https://github.com/project-radius/radius/issues/3701

### CODE REVIEW

File: docs/content/operations/platforms/kubernetes/_index.md [hunk 1]
```diff
@@ -91,6 +91,10 @@ eksctl create cluster --name my-cluster --region region-code --fargate
 Once deployed and your kubectl context has been set as your default, you can run the following to create a Radius environment and install the control plane:
 
 ```bash
+# Note: make sure to provide a valid (alphanumeric with hyphens)
+# environment name, since the default will not be valid.
+
+# Enter an environment name [arn:aws:eks:region:account:cluster/cluster-name]: cluster-name
```
Comment (AaronCrawfis) [inline docs/content/operations/platforms/kubernetes/_index.md]:
```suggestion
# Note: The default environment name for EKS is invalid with current env name requirements
# As part of the init prompts, provide a custom name with only alphanumeric/hyphen characters

# i.e. Enter an environment name [arn:aws:eks:region:account:cluster/mycluster]: mycluster
```

Comment (AaronCrawfis) [inline docs/content/operations/platforms/kubernetes/_index.md]: @willdavsmith do we have an issue to track a CLI update to change the logic for default env name generation to remove/transform unsupported characters?

File: docs/content/operations/platforms/kubernetes/_index.md [hunk 2]
```diff
@@ -91,6 +91,10 @@ eksctl create cluster --name my-cluster --region region-code --fargate
 Once deployed and your kubectl context has been set as your default, you can run the following to create a Radius environment and install the control plane:
 
 ```bash
+# Note: make sure to provide a valid (alphanumeric with hyphens)
```
Comment (AaronCrawfis) [inline docs/content/operations/platforms/kubernetes/_index.md]: We should also add a note to https://radapp.dev/reference/limitations/

**Review by Reshrahim — COMMENTED** [review]
I think we need to make sure this workaround goes into all the sections that asks users to do environment initialization 
1. Tutorial - https://radapp.dev/getting-started/tutorial/init-environment/
2. Environments - https://radapp.dev/operations/environments/

File: docs/content/operations/platforms/kubernetes/_index.md (hunk 1 above)
Comment (willdavsmith) [reply docs/content/operations/platforms/kubernetes/_index.md]: yes, https://github.com/project-radius/radius/issues/3629

**Comment by willdavsmith** [comment]
> I think we need to make sure this workaround goes into all the sections that asks users to do environment initialization
> 
> 1. Tutorial - https://radapp.dev/getting-started/tutorial/init-environment/
> 2. Environments - https://radapp.dev/operations/environments/

IMO, I don't think these need to be updated. Both of them say that that `rad env init kubernetes` should be called with the `-i` flag, and it specifies the name requirements, which is exactly what the workaround is. This issue is specific to the defaults for EKS. What do you think @Reshrahim ?

**Review by AaronCrawfis — DISMISSED** [review]

**Comment by AaronCrawfis** [comment]
> IMO, I don't think these need to be updated. Both of them say that that `rad env init kubernetes` should be called with the `-i` flag, and it specifies the name requirements, which is exactly what the workaround is. This issue is specific to the defaults for EKS. What do you think @Reshrahim ?

An easy solution would be to add the following markdown in these locations:

```md
> Visit the [Kubernetes platform docs]({{< ref kubernetes >}}) for a list of supported clusters and specific cluster requirements.
```

**Comment by Reshrahim** [comment]
> 



> IMO, I don't think these need to be updated. Both of them say that that `rad env init kubernetes` should be called with the `-i` flag, and it specifies the name requirements, which is exactly what the workaround is. This issue is specific to the defaults for EKS. What do you think @Reshrahim ?

Have seen users give their own name for env when skimming through tutorial instructions. Would be good to call this out in these places as well.

File: docs/content/operations/platforms/kubernetes/_index.md [hunk 3]
```diff
@@ -85,12 +85,16 @@ rad env init kubernetes -i --public-endpoint-override 'http://localhost:8080'
 Amazon Elastic Kubernetes Service (Amazon EKS) is a managed service that you can use to run Kubernetes on AWS. Learn how to set up an EKS cluster on the [AWS docs](https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html).
 
 ```bash
-eksctl create cluster --name my-cluster --region region-code --fargate
```
Comment (willdavsmith) [inline docs/content/operations/platforms/kubernetes/_index.md:88]: removed this, focusing on ec2-based eks clusters for now

File: docs/content/operations/environments/_index.md [hunk 4]
```diff
@@ -83,6 +83,8 @@ The following example shows an environment configured with Kubernetes as the tar
 ## How-to: Initialize a new environment
 
 1. Begin by deploying a compatible [Kubernetes cluster]({{< ref kubernetes >}})
+> Visit the [Kubernetes platform docs]({{< ref kubernetes >}}) for a list of supported clusters and specific cluster requirements.
```
Comment (AaronCrawfis) [inline docs/content/operations/environments/_index.md]: This is rendering the note out-of-line with the numbers: 
image

I think you can indent the note to have it line up correctly

Comment (AaronCrawfis) [reply docs/content/operations/environments/_index.md]:
```suggestion
   *Visit the [Kubernetes platform docs]({{< ref kubernetes >}}) for a list of supported clusters and specific cluster requirements.*
```

Looks like an indented blockquote isn't possible. Let's go with an italicized indented paragraph

Comment (willdavsmith) [reply docs/content/operations/environments/_index.md]: this also didn't work for this file. I made this change in the other file and in this one I moved it down just a bit:
image

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**